targettests/execution/mapping_test-1-cpp17.cpp

```diff
@@ -7,7 +7,7 @@
  ******************************************************************************/
 
 // REQUIRES: c++17
-// RUN: nvq++ %cpp_std %s -o %t --target oqc --emulate && CUDAQ_DUMP_JIT_IR=1 %t 2> %t.txt | FileCheck %s && FileCheck --check-prefix=QUAKE %s < %t.txt
```

Comment:
Why are the QIR checks dropped on the floor now?

Reply:
The QIR checks were overly restrictive. There were cases where swapping both result numbers and result strings would result in semantically the same answer, but the QIR checks couldn't verify that. I replaced the QIR checks with answer checks where required.
(Side note: those updates aren't strictly required for this PR, but they will be required if we ever run Adam's pass prior to mapping. I could remove them if you want, but we would probably have to add them back in to support Adam's other changes.)

Reply:
I'm thinking let's leave them in then, but add the new checks as well. When we get to the point they have to come out, we can revisit.

Reply:
> I'm thinking let's leave them in then, but add the new checks as well. When we get to the point they have to come out, we can revisit.

OK, I will add them back in.

Reply:
> > I'm thinking let's leave them in then, but add the new checks as well. When we get to the point they have to come out, we can revisit.
> 
> OK, I will add them back in.

Added back in a5ae54374925ff45ce0dd67b933cee637b381141